### PR TITLE
Add automatic default stress setup

### LINKED ISSUE #87: stress/trunk directory should be removed in favor of the stress_revision feature with a default revision

### PR DESCRIPTION
**Pull request by aboudreault**
So, the stress setup is now automatically done.

@EnigmaCurry 

fixes #87

### CODE REVIEW

**Comment by EnigmaCurry** [comment]
I love that we have the ability to build stress from within the codebase now, but here's how I think it should work:

setup_stress() should be called one time per job, at the beginning. That way we ensure the same exact git SHA of stress is used for the entire run (what if someone makes commits to stress right in the middle of a two stress commands?)

I think there's only three versions of stress we should use (and in order of preference):
* The version that is specified in the job itself.
* The version that is specified in the cluster_config.json
* apache/trunk

This code in this PR will fallback to a random stress folder (that doesn't start with '_') in ~/fab/stress, which may be very old. I want to move away from any static version of stress. If someone wants to use old versions of stress, they should have to specify an old git SHA in the job or cluster_config. Otherwise, we always build the stress again from scratch at the beginning of the job. As an optimization, we can keep a cache of built versions and compare git SHAs to prevent unnecessary building of the same version (much like we do for C* in fab_cassandra.bootstrap())

I'll put some further reiterations of this idea as line items.

File: tool/cstar_perf/tool/benchmark.py [hunk 1]
```diff
@@ -330,8 +329,21 @@ def drop_page_cache():
     if not config.get('docker', False):
         bash(['sync', 'echo 3 > /proc/sys/vm/drop_caches'], user='root')
 
-def setup_stress(stress_revision):
+def setup_stress(stress_revision=None):
```
Comment (EnigmaCurry) [inline tool/cstar_perf/tool/benchmark.py]: I think this method should have no default setting. I think we should instead change the parameter in the definition of stress(): 

    def stress(..., stress_revision='apache/trunk')

File: tool/cstar_perf/tool/benchmark.py [hunk 2]
```diff
@@ -330,8 +329,21 @@ def drop_page_cache():
...
+    is_default_revision = False
+
+    # If no stress_revision specified, ensure there is at least a stress build available
+    if not stress_revision:
+        stress_builds = [b for b in os.listdir(CASSANDRA_STRESS_PATH) if not b.startswith('_')]
+        if stress_builds:
+            path = os.path.join(CASSANDRA_STRESS_PATH, stress_builds[0])
+            logger.info("Using cassandra-stress from '{}'.".format(path))
+            return os.path.join(path, 'tools/bin/cassandra-stress')
+
```
Comment (EnigmaCurry) [inline tool/cstar_perf/tool/benchmark.py]: I don't like this finding an old version of stress in this folder. I think we should use this simply as a cache folder for prior build versions of stress named by it's git SHA.

**Comment by aboudreault** [comment]
I see, I thought you wanted to preserve the same stress build to avoid any misintepretation of users running multiple tests. But yeah, we can do that by specifying the revision in cluster_config.json.

Good point about the commit in the middle of 2 revisions. We should run setup_stress only at the beginning of the bootstrap. 

I'll do these changes.

File: tool/cstar_perf/tool/benchmark.py (hunk 2 above)
Comment (aboudreault) [inline tool/cstar_perf/tool/benchmark.py]: This will be gone

**Comment by EnigmaCurry** [comment]
That's also a good point about expectation of stress not changing between a person's two jobs. I think we can handle that via the GUI with the addition of the stress version in #86 - when they see that option with 'apache/trunk' prefilled in as the default it should be obvious to them that it may be different between runs, and if they want that to be constant, they should fill in a git SHA there.

**Comment by aboudreault** [comment]
yep, sounds good

**Comment by aboudreault** [comment]
@EnigmaCurry pushed  new version of this work. No caching for now, so stress is build on each run.

**Comment by aboudreault** [comment]
@EnigmaCurry since you asked :)

**Comment by aboudreault** [comment]
rebased

**Comment by EnigmaCurry** [comment]
I changed the logic of this, to be stress_revision on the operation, rather than the revision. I also added GUI options to change the default.

**Pull request merged**